forms/FormFieldSchemaTrait.php

```diff
@@ -16,6 +18,34 @@
 trait FormFieldSchemaTrait {
 
 	/**
+	 * The data type backing the field. Represents the type of value the
+	 * form expects to receive via a postback.
+	 *
+	 * The values allowed in this list include:
+	 *
+	 *   - String: Single line text
+	 *   - Hidden: Hidden field which is posted back without modification
+	 *   - Text: Multi line text
+	 *   - HTML: Rich html text
+	 *   - Integer: Whole number value
+	 *   - Decimal: Decimal value
+	 *   - MultiSelect: Select many from source
+	 *   - SingleSelect: Select one from source
+	 *   - Date: Date only
+	 *   - DateTime: Date and time
+	 *   - Time: Time only
+	 *   - Boolean: Yes or no
+	 *   - Custom: Custom type declared by the front-end component. For fields with this type,
+	 *     the component property is mandatory, and will determine the posted value for this field.
+	 *   - Structural: Represents a field that is NOT posted back. This may contain other fields,
+	 *     or simply be a block of stand-alone content. As with 'Custom',
+	 *     the component property is mandatory if this is assigned.
+	 *
+	 * @var string
+	 */
+	protected $schemaDataType;
```

Comment:
Can we add a docblock note saying that the value for this should be set in definition of the FormField subclasses. I'm not sure if that blocks us from using class constants for the values, though.